### PR TITLE
Add new AutoArchive option

### PR DESCRIPTION
**Pull request by Alex1607**
Added a new AutoArchive option which when enabled will automatically archive images in immich which are also archived in google photos

### CODE REVIEW

File: cmd/upload/upload.go [hunk 1]
```diff
@@ -473,6 +478,10 @@ func (app *UpCmd) handleAsset(ctx context.Context, a *browser.LocalAssetFile) er
 			return nil
 		}
 		app.manageAssetAlbum(ctx, ID, a, advice)
+		if app.AutoArchive && a.Archived {
+			a.Archived = true
+			app.Immich.UpdateAsset(ctx, ID, a)
+		}
```
Comment (simulot) [inline cmd/upload/upload.go:484]: I think you can simplify this code.

The AssetUpload method of the immich client must be updated to handle the Archived flag: 
https://github.com/simulot/immich-go/blob/d58d2669ae6f13bbb32ff45fc787e92f0ae4d166/immich/asset.go#L125-L127

Doing so, you'll save one call

Comment (simulot) [reply cmd/upload/upload.go:484]: Also add commit that mention the fix of the #448

**Review by simulot — CHANGES_REQUESTED** [review]

**Review by simulot — APPROVED** [review]

File: cmd/upload/upload.go [hunk 2]
```diff
@@ -161,6 +162,10 @@ func newCommand(ctx context.Context, common *cmd.SharedFlags, args []string, fsO
 		"discard-archived",
 		" google-photos only: Do not import archived photos (default FALSE)", myflag.BoolFlagFn(&app.DiscardArchived, false))
 
+	cmd.BoolFunc(
+		"auto-archive",
+		" google-photos only: Automatically archive photos that are also archived in google photos (default FALSE)", myflag.BoolFlagFn(&app.AutoArchive, true))
```
Comment (krugerk) [inline cmd/upload/upload.go:167]: The description claims the default is false while it looks like the code defaults to true.

Comment (Alex1607) [reply cmd/upload/upload.go:167]: Good catch, I missed that when I updated the default to true

**Pull request merged**